gdx/src/com/badlogic/gdx/utils/XmlReader.java

```diff
@@ -544,9 +544,9 @@ public void remove () {
 
 		public void replaceChild (Element child, Element replacement) {
 			if (children == null) throw new GdxRuntimeException("Element has no children: " + name);
-			final int index = children.indexOf(child, true);
-			if (index == -1) throw new GdxRuntimeException("Element does not contain child: " + child);
-			children.set(index, replacement);
+			if (!children.replaceFirst(child, true, replacement)) {
+				throw new GdxRuntimeException("Element '" + name + "' does not contain child: " + child);
```

Comment:
This looks good if you've tested it; I'm just assuming `name` is available at this point in the code. It looks like an improvement, if so!

Reply:
These lines should probably be replaced as well...
https://github.com/libgdx/libgdx/blob/35afa7986e400f3e6e7b9b8dfb3d0fc3d74a4744/gdx/res/com/badlogic/gdx/utils/XmlReader.rl#L335-L337

Reply:
> I'm just assuming `name` is available at this point in the code.

It is used like this in several palces in the Element class and where I use it in my code it works.

Reply:
replaceAll now return the number of replacements done. This might be even more useful than just returning true or false?

Reply:
Looks good to me! I wonder why the JDK doesn't use that approach... I do like this better!